### PR TITLE
Cult changes

### PR DESCRIPTION
**Pull request by ShadowDeath6**
best group antag tbh

Cultists can now slap a pylon with a shielded robe to recharge it, hopefully fixing the meme of throwing useless robes onto the ground because no more shield. Time to recharge is 20 seconds, could be changed based on feedback

Cultists can now hit a runed wall with their tome to toggle phasability or not. This was done because artificers were just spamming walls and nobody could do anything about it other than artificers.

Phasable walls are now no longer opaque so you can see through them, and they have a slightly modified alpha so it's a bit more obvious that you can walk through them.

fixed bugs: 
harvesters can now see in the dark correctly
harvesters and artificers now correctly corrupt floors on click

now with 100% more actual testing!

:cl: ShadowDeath6
rscadd: Cultists can now hit a runed wall with their tome to make it phasable.
rscadd: Cultists can now recharge shielded robes by using them on a pylon.
rscadd: Soulstones now appear above items on the ground.
rscdel: Constructs can no longer pull people through walls.
bugfix: Harvesters and Artificers can now correctly corrupt floors.
bugfix: Harvesters can now correctly see in the dark.
tweak: Phasable cult walls are now more obvious and are see-through.
/:cl:

### CODE REVIEW

**Comment by GrayRachnid** [comment]
> Phasable walls are now no longer opaque so you can see through them, and they have a slightly modified alpha so it's a bit more obvious that you can walk through them.

why not just make it easier to put down narsie airlocks?

**Comment by Notamaniac** [comment]
The magic missile  'Drag through a cult wall into space' meme still lives it seems.

**Comment by ShadowDeath6** [comment]
Narsie airlocks are already in a good spot right now, I guess I can add artificers being able to make airlocks as well

Magic missle stun is pretty short, certainly not an uncounterable meme

File: code/game/gamemodes/cult/cult_structures.dm [hunk 1]
```diff
@@ -141,6 +141,19 @@
...
 			last_corrupt = world.time + corrupt_delay*2
 
+/obj/structure/cult/pylon/attackby(obj/I, mob/user, params)
+	if(istype(I, /obj/item/clothing/suit/hooded/cultrobes/cult_shield) && iscultist(user))
+		var/obj/item/clothing/suit/hooded/cultrobes/cult_shield/S = I
+		if(S.current_charges >= 3)
+			user <<"<span class='warning'>[I] is already fully charged!</span>"
+			return
+		user <<"<span class='notice'>You hold [I] up to [src]. Hold still...</span>"
+		if(do_mob(user, user, 200))
```
Comment (Super3222) [inline code/game/gamemodes/cult/cult_structures.dm]: The pylon should have a cooldown for this ability.

**Comment by Super3222** [comment]
What's the point of rune metal when you can get an artificer to build your base for you? No, don't do that. They _ARE_ in a good spot right now.

Also a suggestion, make it so the walls don't open when they're right next to a space tile.  You know what I mean?

Also, is it possible that you can make it so that humans can NOT be pulled through walls? That needs working.

Tomes hitting cult walls should make the wall cave open or something. With a cool animation, you know what I mean. I'll put up sprite request if you'd wish to work on it in this PR.

**Comment by ShadowDeath6** [comment]
Until we merge gunhog's thing that lets you get runed metal from the supply talisman, cultists will always struggle to build bases because plasteel is a bitch to come by unless you're science, cargo, or engineering, and even then you get 10 sheets per department plus EVA if you're feeling adventurous. 

There is no good way to make runewalls because they take 2 runed metal which could be better used on literally anything else. 
Ironically, cult has shittier base building power than current clock cult, while clock cult can convert much more easily on the fly.

**Comment by Super3222** [comment]
As long as it's not creating airlocks out of cult magic, artificers are in a good spot.
And this:

> The magic missile 'Drag through a cult wall into space' meme still lives it seems.

Is almost impossible to counter.

**Comment by ShadowDeath6** [comment]
I can make the magic missile stun for less, but if you're right next to an artificer in a cult base you kind of had it coming

**Comment by Super3222** [comment]
You are ignoring the obvious solution. I'll figure it out on my own shortly.

File: code/game/gamemodes/cult/cult_structures.dm [hunk 2]
```diff
@@ -141,6 +141,22 @@
...
+/obj/structure/cult/pylon/attackby(obj/I, mob/user, params)
+	if(istype(I, /obj/item/clothing/suit/hooded/cultrobes/cult_shield) && iscultist(user))
+		var/obj/item/clothing/suit/hooded/cultrobes/cult_shield/S = I
+		if(S.current_charges >= 3)
+			user <<"<span class='warning'>[I] is already fully charged!</span>"
+			return
+		if(cooldowntime > world.time)
+			user << "<span class='cultitalic'>The magic in [src] is weak, it will be ready to use again in [getETA()].</span>"
+			return
+		if(src && !qdeleted(src) && anchored && Adjacent(user) && !user.incapacitated() && iscultist(user) && cooldowntime <= world.time)
```
Comment (Super3222) [inline code/game/gamemodes/cult/cult_structures.dm]: Most of these are handled in if(do_mob) and you really don't need to worry about them.

If you included the actual pylon in the arguemeent, you wouldn't have to go through all of these checks. Instead this line should be.
`if(anchored && Adjacent(user) && cooldowntime <= world.time)`

Whether they are a cultist or not was already pointed out ...

File: code/game/gamemodes/cult/cult_structures.dm [hunk 3]
```diff
@@ -141,6 +141,22 @@
 		else
 			last_corrupt = world.time + corrupt_delay*2
 
+/obj/structure/cult/pylon/attackby(obj/I, mob/user, params)
+	if(istype(I, /obj/item/clothing/suit/hooded/cultrobes/cult_shield) && iscultist(user))
```
Comment (Super3222) [inline code/game/gamemodes/cult/cult_structures.dm:145]: ... over here.

I'm working kind of backwards but, you've already made a check for whether they're a cultist or not. No apparent reason to duplicate it.

**Comment by ghost** [comment]
May want to allow harvesters to create runed metal and convert airlocks into cult airlocks by attacking them

**Comment by ShadowDeath6** [comment]
runed metal wouldn't be canonically correct because they're supposed to kill everyone and take them to nar'sie, it's why they're not cultists

I'm not even sure if they can use cult airlocks

**Comment by ShadowDeath6** [comment]
added the stuff from https://github.com/yogstation13/yogstation/pull/477 also

**Comment by Super3222** [comment]
I'm pretty sure anyone anyone without a PASSDOOR can open an airlock.

**Comment by ShadowDeath6** [comment]
Harvesters aren't actually considered cult though

File: code/game/gamemodes/cult/cult_structures.dm (hunk 3 above)
Comment (oisin100) [inline code/game/gamemodes/cult/cult_structures.dm:145]: I fail to understand what you mean Super, Could you explain? This code seems perfectly fine.

**Comment by oisin100** [comment]
> Hit wall with tome
> Instant mass driver with no cooldown

File: code/game/turfs/simulated/walls_misc.dm [hunk 4]
```diff
@@ -28,6 +28,40 @@
...
+			sleep(10)
+			phasable--
+		src.density = 1
+		src.alpha = initial(src.alpha)
+		src.opacity = 1
+	return
+
+/turf/closed/wall/mineral/cult/attackby(obj/item/weapon/W, mob/user, params)
+	if(istype(W, /obj/item/weapon/tome) && iscultist(user))
+		if(src.density == 1)
```
Comment (oisin100) [inline code/game/turfs/simulated/walls_misc.dm:53]: Whats with this use of `src` ?
Also, `if(density)` works just as well as `if(src.density == 1)`

**Comment by ShadowDeath6** [comment]
> hit wall with tome
> instant mass driver

what

**Comment by oisin100** [comment]
Well, Air wont go through it, But you can easily throw people through it.

**Comment by oisin100** [comment]
Wait, Apparently TG removed throwing.

File: code/game/gamemodes/cult/cult_structures.dm (hunk 3 above)
Comment (Super3222) [inline code/game/gamemodes/cult/cult_structures.dm:145]: He had another check for `iscultist(user))` later in the proc block. I was showing him why he didn't need it.

**Comment by Super3222** [comment]
You need to make a check for space tiles.

**Comment by ShadowDeath6** [comment]
if the cult wants to breach their own base it's their prerogative, I'm not isolating them to the station like swarmers

**Comment by oisin100** [comment]
^ Besides, Air wont pass through.

**Comment by ShadowDeath6** [comment]
thank you ;_;

**Pull request merged**